platform/security/idp/security-idp-client/src/main/java/org/codice/ddf/security/idp/client/LogoutRequestService.java

```diff
@@ -236,6 +260,104 @@ private Response getSamlpRedirectLogoutRequest(String relayState, LogoutRequest
     return ok.build();
   }
 
+  private String extractSubject(HttpSession httpSession) {
+    return Stream.of(httpSession.getAttribute(SecurityConstants.SAML_ASSERTION))
+        .map(SecurityTokenHolder.class::cast)
+        .map(SecurityTokenHolder::getRealmTokenMap)
+        .map(Map::values)
+        .flatMap(Collection::stream)
+        .map(this::extractSubject)
+        .filter(Objects::nonNull)
+        .map(SubjectUtils::getName)
+        .findFirst()
+        .orElse(null);
+  }
+
+  private Subject extractSubject(SecurityToken securityToken) {
+    try {
+      return securityManager.getSubject(securityToken);
+    } catch (SecurityServiceException e) {
+      LOGGER.debug("Error extracting subject from security token", e);
+      return null;
+    }
+  }
+
+  @POST
+  @Consumes({"text/xml", "application/soap+xml"})
+  public Response soapLogoutRequest(InputStream body, @Context HttpServletRequest request) {
+    XMLObject xmlObject;
+    try {
+      String foo = IOUtils.toString(body, StandardCharsets.UTF_8);
+      SOAPPart soapMessage = SamlProtocol.parseSoapMessage(foo);
+
+      xmlObject =
+          SamlProtocol.getXmlObjectFromNode(soapMessage.getEnvelope().getBody().getFirstChild());
+      if (!(xmlObject instanceof LogoutRequest)) {
+        LOGGER.info(UNABLE_TO_PARSE_LOGOUT_REQUEST);
+        if (LOGGER.isDebugEnabled()) {
+          LOGGER.debug(
+              "Type of object is {}", xmlObject == null ? "null" : xmlObject.getSchemaType());
+        }
+        return buildLogoutResponse(UNABLE_TO_PARSE_LOGOUT_REQUEST);
```

Comment:
All of these responses are wrong, as `buildLogoutResponse` in this class returns with a status of `303`. Instead, they should be 
> `200 OK` and include a SAML-specified `<samlp:Status>` element in the SAML response within the SOAP body